api/tests/Unit/PoolCandidateAuthorizationScopeTest.php

```diff
@@ -0,0 +1,186 @@
+
+        $this->candidateSubmitted2A = PoolCandidate::factory()
+            ->for($this->user2)
+            ->for($this->poolA)
+            ->create([
+                'pool_candidate_status' => PoolCandidateStatus::NEW_APPLICATION->name,
+            ]);
+    }
+
+    // a guest should be able to view no candidates
+    public function testViewAsGuest(): void
+    {
+        Auth::shouldReceive('user')
+            ->andReturn(null);
+
+        $candidateIds = PoolCandidate::authorizedToView()->get()->pluck('id');
+        assertEqualsCanonicalizing([], $candidateIds->toArray());
+    }
+
+    // an applicant should be able to view their own candidates
+    public function testViewAsApplicant(): void
+    {
+        Auth::shouldReceive('user')
+            ->andReturn($this->user1);
+
+        // just the draft and submitted candidates for user 1
+        $candidateIds = PoolCandidate::authorizedToView()->get()->pluck('id');
+        assertEqualsCanonicalizing([
+            $this->candidateDraft1A->id,
+            $this->candidateSubmitted1B->id,
+        ], $candidateIds->toArray());
+    }
+
+    // a pool operator should be able to view submitted candidates in their team's pools
+    public function testViewAsPoolOperator(): void
+    {
+        Auth::shouldReceive('user')
+            ->andReturn(
+                User::factory()
+                    ->asPoolOperator($this->teamA->name)
+                    ->create()
+            );
+
+        // the pools in team A only have one submitted candidate
+        $candidateIds = PoolCandidate::authorizedToView()->get()->pluck('id');
+        assertEqualsCanonicalizing([
+            $this->candidateSubmitted2A->id,
+        ], $candidateIds->toArray());
+    }
+
+    // a request responder should be able to view submitted candidates in any team
+    public function testViewAsRequestResponder(): void
+    {
+        Auth::shouldReceive('user')
+            ->andReturn(
+                User::factory()
+                    ->asRequestResponder()
+                    ->create()
+            );
+
+        // there is one submitted candidate in each team's pool
+        $candidateIds = PoolCandidate::authorizedToView()->get()->pluck('id');
+        assertEqualsCanonicalizing([
+            $this->candidateSubmitted1B->id,
+            $this->candidateSubmitted2A->id,
+        ], $candidateIds->toArray());
+    }
+
+    // community manager role doesn't grant any candidate permissions - nothing viewable
+    public function testViewAsCommunityManager(): void
+    {
+        Auth::shouldReceive('user')
+            ->andReturn(
+                User::factory()
+                    ->asCommunityManager()
+                    ->create()
+            );
+
+        $candidateIds = PoolCandidate::authorizedToView()->get()->pluck('id');
+        assertEqualsCanonicalizing([], $candidateIds->toArray());
+    }
+
+    // a platform admin should be able to view submitted candidates in any team
```

Comment:
Technically, this goes against the spec.

Reply:
Thanks, I've updated the spec in this case.